db/db_impl/db_impl.cc

```diff
@@ -2901,6 +2901,14 @@ Iterator* DBImpl::NewIterator(const ReadOptions& read_options,
   }
   // if iterator wants internal keys, we can only proceed if
   // we can guarantee the deletes haven't been processed yet
+  if (read_options.iter_start_seqnum > 0 &&
+      !iter_start_seqnum_deprecation_warned_) {
+    ROCKS_LOG_WARN(
+        immutable_db_options_.info_log,
+        "iter_start_seqnum is deprecated, will be removed in a future release. "
+        "Please try using user-defined timestamp instead.");
+    iter_start_seqnum_deprecation_warned_ = true;
```

Comment:
Sorry for this late comment: will there be a race condition for accessing `iter_start_seqnum_deprecation_warned_` here?

Reply:
Good point, updated with using `atomic_bool.exchange()`.